include/boost/geometry/algorithms/detail/overlay/get_turns.hpp

```diff
@@ -101,6 +101,105 @@ struct no_interrupt_policy
+                          CircularIterator circular_iterator,
+                          Point const& previous, Point const& current,
+                          RobustPolicy const& robust_policy)
+      : m_section(section)
+      , m_index(index)
+      , m_previous_point(previous)
+      , m_current_point(current)
+      , m_circular_iterator(circular_iterator)
+      , m_point_retrieved(false)
+      , m_robust_policy(robust_policy)
+    {
+    }
+
+    inline bool is_first() const
+    {
+        return !IsAreal && m_section.is_non_duplicate_first && m_index == m_section.begin_index;
+    }
+
+    inline std::size_t size() const
+    {
+        return IsAreal ? 3
+            : m_section.is_non_duplicate_last && m_index + 1 >= m_section.end_index ? 2 : 3;
+    }
+
+    inline Point const& at(std::size_t index) const
+    {
+        switch (index)
+        {
+            case 0 : return m_previous_point;
+            case 1 : return m_current_point;
+            case 2 : return get_next_point();
+            default : return m_previous_point;
+        }
+    }
+
+private :
+    inline Point const& get_next_point() const
+    {
+        if (! m_point_retrieved)
+        {
+            advance_to_non_duplicate_next(m_current_point, m_circular_iterator);
+            m_point = *m_circular_iterator;
+            m_point_retrieved = true;
+        }
+        return m_point;
+    }
+
+    inline void advance_to_non_duplicate_next(Point const& current, CircularIterator& circular_iterator) const
+    {
+        typedef typename robust_point_type<Point, RobustPolicy>::type robust_point_type;
+        robust_point_type current_robust_point;
+        robust_point_type next_robust_point;
+        geometry::recalculate(current_robust_point, current, m_robust_policy);
+        geometry::recalculate(next_robust_point, *circular_iterator, m_robust_policy);
+
+        // To see where the next segments bend to, in case of touch/intersections
+        // on end points, we need (in case of degenerate/duplicate points) an extra
+        // iterator which moves to the REAL next point, so non duplicate.
+        // This needs an extra comparison (disjoint).
+        // (Note that within sections, non duplicate points are already asserted,
+        //   by the sectionalize process).
+
+        // So advance to the "non duplicate next"
+        // (the check is defensive, to avoid endless loops)
+        std::size_t check = 0;
+        while(! detail::disjoint::disjoint_point_point
+                (
+                    current_robust_point, next_robust_point
+                )
+            && check++ < m_section.range_count)
+        {
+            circular_iterator++;
+            geometry::recalculate(next_robust_point, *circular_iterator, m_robust_policy);
+        }
+    }
+
+    Section const& m_section;
+    signed_size_type m_index;
+    Point const& m_previous_point;
+    Point const& m_current_point;
```

Comment:
And here points are stored as references. What is the reason behind the inconsistency?